app/assets/javascripts/angular/controllers/topics_ctrl.coffee

```diff
@@ -15,6 +15,15 @@ angular.module "lean-coffee"
       if confirm('Are you sure you want to delete this?')
         topicsService.delete(topic)
 
+    @archive = (topic) =>
+      if confirm('Are you sure you want to archive this?')
+        topicsService.archive(topic)
+
+    @archiveAll = =>
+      if confirm('Are you sure you want to archive all discussed topics?')
+        topicsService.archiveAll()
+        topicsService.reload()
```

Comment:
When a browser is the one that makes an ajax call, [this `socket_id` setting](https://github.com/chadoh/lean-coffee/blob/rooms/app/assets/javascripts/application.coffee#L32) causes it to ignore the resulting Pusher update. It itself is updated the traditional way, using the payload of the ajax request.

At least, that's how it works now. I'm experimenting with removing the `socket_id` so that all of the pages are updated the same way. But right now, that causes nasty infinite loops of ajax requests! 

For some reason, on page load, an `update` request is fired for _every single topic on the page_. ***FOUR TIMES**. If you have two topics on the page, Angular sends off 8 ajax requests right after page load! No idea why yet. But if I naively remove this `socket_id`, then each of those kicks off another (which is also mysterious), and :boom: